src/Sarif.WorkItems/SarifWorkItemFiler.cs

```diff
@@ -18,44 +16,29 @@ public class SarifWorkItemFiler : IDisposable
         /// <summary>
         /// Initializes a new instance of the <see cref="SarifWorkItemFiler"> class.</see>
         /// </summary>
-        /// <param name="filingClient">
-        /// A client for communicating with a work item filing host (for example, GitHub or Azure DevOps).
+        /// <param name="filingUri">
+        /// The uri to the remote filing host.
         /// </param>
         /// <param name="filingContext">
         /// A starting context object that configures the work item filing operation. In the
         /// current implementation, this context is copied for each SARIF file (if any) split
         /// from the input log and then further elaborated upon.
         /// </param>
-        public SarifWorkItemFiler(Uri filingUri)
+        public SarifWorkItemFiler(Uri filingUri = null, SarifWorkItemContext filingContext = null)
         {
-            if (filingUri == null) { throw new ArgumentOutOfRangeException(nameof(filingUri)); };
+            this.FilingContext = filingContext ?? new SarifWorkItemContext { HostUri = filingUri };
+            filingUri = filingUri ?? this.FilingContext.HostUri;
 
-            this.FilingContext = new SarifWorkItemContext
+            if (filingUri == null) { throw new ArgumentNullException(nameof(filingUri)); };
+
+            if (filingUri != this.FilingContext.HostUri)
             {
-                HostUri = filingUri
-            };
+                throw new InvalidOperationException();
```

Comment:
Isn't this where you want to use that `InconsistentHostUris` resource string you just added?